### PR TITLE
BUG: sorting with large float and multiple columns incorrect

### LINKED ISSUE #14922: BUG: sorting with large float and multiple columns incorrect
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

int_values = (2, -9223372036854775808)
float_values = (2.0, -1.797693e308)

df = pd.DataFrame(dict(int=int_values,
                       float=float_values), columns=("int", "float"))

print("df")
print(df)
print()

print("sort by int")
print(df.sort_values(["int"]))
print()

print("sort by float")
print(df.sort_values(["float"]))
print()

print("sort by (int, float)")
print(df.sort_values(["int", "float"]))
```

produces

```
$ python wrong.py
df
                   int          float
0                    2   2.000000e+00
1 -9223372036854775808 -1.797693e+308

sort by int
                   int          float
1 -9223372036854775808 -1.797693e+308
0                    2   2.000000e+00

sort by float
                   int          float
1 -9223372036854775808 -1.797693e+308
0                    2   2.000000e+00

sort by (int, float)
                   int          float
0                    2   2.000000e+00
1 -9223372036854775808 -1.797693e+308
```

As you can see the first two calls (only sorting by `int` or by `float`) sort correctly whereas `df.sort_values(["int", "float"])`  does not seem to sort at all. Row with index `1` should appear first, and then row with index `0`.

Using `pandas 0.17.0` I get the  correct result:

```
                   int          float
1 -9223372036854775808 -1.797693e+308
0                    2   2.000000e+00
```

####

This issue was discussed at http://stackoverflow.com/questions/41225286/sort-by-broken-in-pandas-0-18-0
    
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS                                                                                                                                                         [1/193]
------------------
commit: None
python: 2.7.12.final.0
python-bits: 64
OS: Darwin
OS-release: 15.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: None.None

pandas: 0.19.1
nose: None
pip: 7.1.2
setuptools: 18.2
Cython: None
numpy: 1.10.4
scipy: None
statsmodels: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
boto: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by uweschmitt**
Fixes https://github.com/pandas-dev/pandas/issues/14922
    
Having the `int` equivalent of `NaT` in an `int64` column caused wrong sorting because this special value was considered as "missing value".

 - [ ] closes #xxxx
 - [X] tests added / passed
 - [X] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

**Comment by uweschmitt** [comment]
@jreback please review https://github.com/pandas-dev/pandas/pull/14944/commits/03699c67a5953882e025ea6eb2607ea3053c9337, I used rebase and now the changes do not show up in github as expected.

File: pandas/tests/frame/test_sorting.py [hunk 1]
```diff
@@ -6,11 +6,12 @@
...
 from pandas.compat import lrange
 from pandas import (DataFrame, Series, MultiIndex, Timestamp,
-                    date_range)
+                    date_range, NaT)
 
 from pandas.util.testing import (assert_series_equal,
                                  assert_frame_equal,
-                                 assertRaisesRegexp)
+                                 assertRaisesRegexp,
+                                 is_sorted)
```
Comment (jreback) [inline pandas/tests/frame/test_sorting.py]: you don't need to import the is_sorted

**Comment by jreback** [comment]
can you add a similar tests which uses datetimes and a NaT as well (just to verify that it doesn't break).

further pls check both ascending and descending.

File: pandas/tests/frame/test_sorting.py [hunk 2]
```diff
@@ -491,3 +492,18 @@ def test_frame_column_inplace_sort_exception(self):
 
         cp = s.copy()
         cp.sort_values()  # it works!
+
+    def test_sort_nat_values_in_int_column(self):
+
+        # GH 14922, sorting with large float and multiple columns incorrect
+        int_values = (2, int(NaT))
```
Comment (jreback) [inline pandas/tests/frame/test_sorting.py]: use np.iinfo(np.int64).min instead (and NaT) is already an int

Comment (uweschmitt) [inline pandas/tests/frame/test_sorting.py]: Thought `NaT` is implicitly indicating what went wrong...

**Comment by jreback** [comment]
can you add a whatsnew note (0.20.0, bug fixes)

File: pandas/tests/frame/test_sorting.py [hunk 3]
```diff
@@ -491,3 +491,56 @@ def test_frame_column_inplace_sort_exception(self):
...
+        df_sorted = df.sort_values(["int", "float"], na_position="first")
+        assert_frame_equal(df_sorted, df_reversed)
+
+        # reverse sorting order
+        df_sorted = df.sort_values(["int", "float"], ascending=False)
+        assert_frame_equal(df_sorted, df)
+
+        # and now check if NaT is still considered as "na" for datetime64
+        # columns:
+        df = DataFrame(dict(int=int_values, float=float_values),
```
Comment (jreback) [inline pandas/tests/frame/test_sorting.py]: this is a not-used line

File: pandas/tests/frame/test_sorting.py [hunk 4]
```diff
@@ -491,3 +491,56 @@ def test_frame_column_inplace_sort_exception(self):
...
+                       columns=["int", "float"])
+
+        df = DataFrame(dict(datetime=[Timestamp("2016-01-01"), NaT],
+                            float=float_values), columns=["datetime", "float"])
+
+        # check if the dtype is datetime64[ns]:
```
Comment (jreback) [inline pandas/tests/frame/test_sorting.py]: remove this line

File: pandas/tests/frame/test_sorting.py [hunk 5]
```diff
@@ -491,3 +491,52 @@ def test_frame_column_inplace_sort_exception(self):
...
+        # reverse sorting order
+        df_sorted = df.sort_values(["int", "float"], ascending=False)
+        assert_frame_equal(df_sorted, df)
+
+        # and now check if NaT is still considered as "na" for datetime64
+        # columns:
+        df = DataFrame(dict(datetime=[Timestamp("2016-01-01"), NaT],
+                            float=float_values), columns=["datetime", "float"])
+
+        assert df["datetime"].dtypes == np.dtype("datetime64[ns]"),\
```
Comment (jreback) [inline pandas/tests/frame/test_sorting.py]: this line here is not needed (the assert of datetime)

Comment (uweschmitt) [inline pandas/tests/frame/test_sorting.py]: done.

**Comment by uweschmitt** [comment]
Ok now ?

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**